src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -422,14 +476,214 @@ export class DeployCommand implements Command {
+          isSecure: deploymentParameter.isSecure,
+          parameterType: deploymentParameter.parameterType,
+        };
+        updatedDeploymentParameters.push(updatedDeploymentParameter);
+      }
+    }
+
+    return [
+      bicepDeploymentParametersResponse.parametersFileName,
+      updatedDeploymentParameters,
+    ];
+  }
+
+  private async updateParametersFile(
+    _context: IActionContext,
+    documentPath: string,
+    parametersFileExists: boolean,
+    parametersFileName: string
+  ) {
+    let placeholder: string;
+    let parametersFileUpdateOptionString = "None";
+    let parametersFileUpdateOption: ParametersFileUpdateOption;
+    if (parametersFileExists) {
+      parametersFileUpdateOptionString = "Update";
+      parametersFileUpdateOption = ParametersFileUpdateOption.Update;
+      placeholder = `Update ${parametersFileName} with values used in this deployment?`;
+    } else {
+      const folderContainingSourceFile = path.dirname(documentPath);
+      const parametersFilePath = path.join(
+        folderContainingSourceFile,
+        parametersFileName
+      );
+      if (fse.existsSync(parametersFilePath)) {
+        parametersFileUpdateOptionString = "Overwrite";
+        parametersFileUpdateOption = ParametersFileUpdateOption.Overwrite;
+        placeholder = `File ${parametersFileName} already exists. Do you want to overwrite it?`;
+      } else {
+        parametersFileUpdateOptionString = "Create";
+        parametersFileUpdateOption = ParametersFileUpdateOption.Create;
+        placeholder = `Create parameters file from values used in this deployment?`;
+      }
+    }
+
+    const result: IAzureQuickPickItem = await _context.ui.showQuickPick(
+      this._yesNoQuickPickItems,
+      {
+        canPickMany: false,
+        placeHolder: placeholder,
+        suppressPersistence: true,
+      }
+    );
+
+    _context.telemetry.properties.parametersFileUpdateOption =
+      parametersFileUpdateOptionString;
+    if (result == this._yes) {
+      return parametersFileUpdateOption;
+    } else {
+      return ParametersFileUpdateOption.None;
+    }
+  }
+
+  private async selectValueForParameterOfTypeExpression(
+    _context: IActionContext,
+    paramName: string,
+    paramValue: string | undefined
+  ) {
+    const quickPickItems: IAzureQuickPickItem[] = [];
+    if (paramValue) {
+      const useExpressionValue: IAzureQuickPickItem = {
+        label: localize(
+          "useExpressionValue",
+          `Use value of expression "${paramValue}"`
+        ),
+        data: undefined,
+      };
+      quickPickItems.push(useExpressionValue);
+    }
+    const enterNewValue: IAzureQuickPickItem = {
+      label: localize(
+        "enterNewValueForParameter",
+        `Enter value for "${paramName}"`
+      ),
+      data: undefined,
+    };
+    quickPickItems.push(enterNewValue);
+    const leaveBlank: IAzureQuickPickItem = {
+      label: localize("leaveBlank", `"" (leave blank)`),
```

Comment:
I think we should remove the "leave blank" option.  They can use the "Enter value" option instead.  Plus, blank ("") only applies to string parameters, but you're showing it anyway.

Reply:
@ucheNkadiCode, I think we should remove  "leave blank" to avoid confusion. User can choose - option (2) from below and hit enter to get same experience. Do you have any concerns in removing it?
![image](https://user-images.githubusercontent.com/30270536/172180423-25c0247e-2ae7-4adb-992b-e4b17d0d650e.png)

Reply:
I agree, let's remove it for simplicity sake

Reply:
Removed it here:
https://github.com/Azure/bicep/commit/5beb12ea871df5d4579492bc7d6c83043d6258b3